cmd/rpcdaemon/commands/eth_block.go

```diff
@@ -14,7 +14,9 @@ import (
 // GetBlockByNumber see https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getblockbynumber
 // see internal/ethapi.PublicBlockChainAPI.GetBlockByNumber
 func (api *APIImpl) GetBlockByNumber(ctx context.Context, number rpc.BlockNumber, fullTx bool) (map[string]interface{}, error) {
-	tx, err := api.dbReader.Begin(ctx)
+	// TODO the next line used to be "tx, err := api.dbReader.Begin(ctx)" When it changed, there was concern that it would have
+	// negative impact on performance. Revisit performance of using this type of database transaction
+	tx, err := api.db.Begin(ctx, nil, false)
```

Comment:
@AskAlexSharov @tjayrush , I'm still feeling uncomfortable with this change: 
- because it works by accident. For example in next lines `tx` object used as: `ReadBlockByNumber(tx)`. If you go inside `ReadBlockByNumber` you can find `!errors.Is(err, ethdb.ErrKeyNotFound)` - but ethdb.Tx doesn't return this error - and it satisfy `DatabaseReader` by accident - (I faced this problem last week - when passed Tx to place which expected Db - and I promised to change tx API - rename Get to GetOne or even remove the method).
- we can merge this PR if it doesn't break things and solve problem of localDb use of RPCDaemon. But I will make PR now which will allow open Read tx by TxDb/ObjectDb - and will ask you to use it. Then will take a look how hard to change LmdbTx to not satisfy rawdb.DatabaseReader interface.

Reply:
@AlexeyAkhunov , @tjayrush here you go guys, sorry for the mess: https://github.com/ledgerwatch/turbo-geth/pull/1287

Reply:
And the one about protection: https://github.com/ledgerwatch/turbo-geth/pull/1288
hope some day we will drop 1 level of abstraction.

Reply:
Trying to catch up. Looks the PR was merged, so I guess you fixed it.